cirrus/server/tests/test_feature_manifest.py

```diff
@@ -1,196 +1,181 @@
-                        "value": {"enabled": True},
-                    },
-                    "featureId": feature_name,
-                    "slug": "experiment_slug",
-                }
-            },
-            "enrollments": [
-                {
-                    "slug": "experiment_slug",
-                    "status": {
-                        "Enrolled": {
-                            "branch": "treatment",
-                            "enrollment_id": "enrollment_id",
-                            "reason": "Qualified",
-                        }
-                    },
-                }
-            ],
-            "events": [
-                {
-                    "branch_slug": "treatment",
-                    "change": "Enrollment",
-                    "enrollment_id": "enrollment_id",
-                    "experiment_slug": "experiment_slug",
-                    "reason": None,
-                }
-            ],
-        }
 
-        result = self.fml.compute_feature_configurations(enrolled_partial_configuration)
+def test_invalid_fml_path():
+    with pytest.raises(FmlError) as e:
+        FeatureManifestLanguage("invalid_fml_path.txt", channel)
+    assert str(e.value).startswith("FmlError.InvalidPath")
 
-        self.assertEqual(
-            result,
-            {
-                "example-feature": {
-                    "enabled": expected_feature_status,
-                    "something": "wicked",
-                }
-            },
-        )
-
-    def test_compute_feature_configurations_invalid_key_merge_errors(self):
-        enrolled_partial_configuration = {
-            "enrolledFeatureConfigMap": {
-                "example-feature": {
-                    "branch": "treatment",
-                    "feature": {
-                        "featureId": "example-feature",
-                        "value": {"enabled1": True},  # invalid
-                    },
-                    "featureId": "example-feature",
-                    "slug": "experiment-slug",
-                }
-            },
-            "enrollments": [],
-            "events": [],
-        }
 
-        result = self.fml.compute_feature_configurations(enrolled_partial_configuration)
-
-        self.assertEqual(
-            result, {"example-feature": {"enabled": False, "something": "wicked"}}
-        )  # returns default config
-        self.assertEqual(len(self.fml.merge_errors), 1)
-        assert isinstance(self.fml.merge_errors[0], FmlError)
-
-    def test_compute_feature_configurations_targeting_doesnt_match(self):
-        bucket_config = {
-            "randomizationUnit": "user_id",
-            "count": 100,
-            "namespace": "",
-            "start": 1,
-            "total": 100,
-        }
+def test_invalid_channel():
+    with pytest.raises(FmlError) as e:
+        FeatureManifestLanguage(fml_path, "invalid_channel")
+    assert str(e.value).startswith("FmlError.InvalidChannelError")
```

Comment:
Does this pass, with the assert outside of the context for where the variable ```e``` is declared?

Reply:
it does work, but it should be intended, good point